presto-main/src/main/java/com/facebook/presto/operator/WindowOperator.java

```diff
@@ -441,6 +452,7 @@ private void sortPagesIndexIfNecessary()
                 startPosition = endPosition;
             }
         }
+        windowInfo.addIndex(pagesIndex);
```

Comment:
It is a bit odd that `windowInfo.addIndex(pagesIndex)` is part of `sortPagesIndexIfNecessary()` (method name is unrelated). Maybe add another method, `finishPagesIndex()`:
```
void finishPagesIndex() {
  sortPagesIndexIfNecessary();
  windowInfo.addIndex(pagesIndex);
}
```
or at least add a comment